File: src/main/java/com/efedotov/vaultly/dto/file/FileDto.java
package com.efedotov.vaultly.dto.file;

import lombok.Data;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
public class FileDto {
    private UUID id;
    private String name;
    private String originalName;
    private Long size;
    private String mimeType;
    private String s3Url;
    private Boolean isEncrypted;
    private Boolean isPublic;
    private Boolean isNote;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private UUID folderId;
    private String folderName;
}

File: lib/server/model/file/file_dto/file_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'file_dto.freezed.dart';
part 'file_dto.g.dart';

@freezed
abstract class FileDto with _$FileDto {
  const factory FileDto({
    String? id,
    required String name,
    required String originalName,
    required int size,
    required String mimeType,
    String? s3Url,
    bool? isEncrypted,
    bool? isPublic,
    @JsonKey(name: 'createdAt') DateTime? createdAt,
    String? folderId,
    String? folderName,
  }) = _FileDto;

  factory FileDto.fromJson(Map<String, dynamic> json) =>
      _$FileDtoFromJson(json);
}
